14/umbraco-cms/tutorials/creating-a-backoffice-api/versioning-your-api.md

```diff
@@ -0,0 +1,199 @@
+    public IActionResult GetAllItems(int skip = 0, int take = 10)
+    {
+        // ...
+    }
+
+    [HttpGet("{id:guid}")]
+    [MapToApiVersion("1.0")]
+    public IActionResult GetItem(Guid id)
+    {
+        // ...
+    }
+
+    [HttpPost]
+    [MapToApiVersion("1.0")]
+    public IActionResult CreateItem(string value)
+    {
+        // ...
+    }
+
+    [HttpPut("{id:guid}")]
+    [MapToApiVersion("1.0")]
+    public IActionResult UpdateItem(Guid id, string value)
+    {
+        // ...
+    }
+
+    [HttpDelete("{id:guid}")]
+    [MapToApiVersion("1.0")]
+    public IActionResult DeleteItem(Guid id)
+    {
+        // ...
+    }
+
+    [HttpGet("{id:guid}")]
+    [MapToApiVersion("2.0")]
+    public IActionResult GetItemV2(Guid id)
+    {
+        MyItem? item = AllItems.FirstOrDefault(item => item.Id == id);
+
+        return item is not null
+            ? Ok(item)
+            : OperationStatusResult(
+                MyItemOperationStatus.NotFound,
+                builder => NotFound(
+                    builder
+                        .WithTitle("The item was not found")
+                        .WithDetail("The item with the given ID did not exist.")
+                        .Build()
+                )
+            );
+    }
+
+    [HttpPost]
+    [MapToApiVersion("2.0")]
+    public IActionResult CreateItemV2(string value)
+    {
+        var newItem = new MyItem(value);
+        AllItems.Add(newItem);
+        return CreatedAtId<MyItemApiController>(
+            ctrl => nameof(ctrl.GetItemV2),
+            newItem.Id
+        );
+    }
+}
+```
+
+{% endcode %}
+
+Version 2.0 of the "get" and "create" endpoints - `GetItemV2` and `CreateItemV2` respectively, are added with the code above. The rest of the endpoints remain version 1.0 only.
+
+{% hint style="info" %}
+The version 2.0 endpoints are routed under `/umbraco/management/api/v2/`.
+{% endhint %}
+
+In the example above, the version 2.0 actions are added to the same API controller as their version 1.0 counterparts. If you prefer, they can be added to a new API controller instead. See the examples below:
+
+{% code title="MyItemApiController.cs" %}
+
+```csharp
+[ApiVersion("1.0")]
+public class MyItemApiController : ManagementApiControllerBase
+{
+    [HttpGet]
+    [MapToApiVersion("1.0")]
+    public IActionResult GetAllItems(int skip = 0, int take = 10)
+    {
+        // ...
+    }
+
+    [HttpGet("{id:guid}")]
+    [MapToApiVersion("1.0")]
+    public IActionResult GetItem(Guid id)
+    {
+        // ...
+    }
+
+    [HttpPost]
+    [MapToApiVersion("1.0")]
+    public IActionResult CreateItem(string value)
+    {
+        // ...
+    }
+
+    [HttpPut("{id:guid}")]
+    [MapToApiVersion("1.0")]
+    public IActionResult UpdateItem(Guid id, string value)
+    {
+        // ...
+    }
+
+    [HttpDelete("{id:guid}")]
+    [MapToApiVersion("1.0")]
+    public IActionResult DeleteItem(Guid id)
+    {
+        // ...
+    }
+}
+```
+
+{% endcode %}
+
+..and:
```

Comment:
Not sure if what I've suggested here is 100% correct, but I'm not a big fan of the standalone "..and:" 😅 

```suggestion
With the version 1.0 actions added in a controller sampled above, the version 2.0 actions are added in a new controller, as shown below.
```